File: src/Web/WebSPA/src/app/core/models/employer-application-dto.ts
import { ApplicationListItemDto } from './application-list-item-dto';

export type CurrentCvStatus = 'Available' | 'Missing' | 'ProfileMissing' | 'ProfileReferenceMissing';

export interface EmployerApplicationDto extends ApplicationListItemDto {
  readonly candidateId: string;
  readonly candidateName: string | null;
  readonly coverLetter: string | null;
  readonly currentCvUrl: string | null;
  readonly cvStatus: CurrentCvStatus;
}

File: src/Services/ApplicationService/Application/DTOs/EmployerApplicationDto.cs
using ApplicationService.Domain.Enums;

namespace ApplicationService.Application.DTOs;

public enum CurrentCvStatus { Available, Missing, ProfileMissing, ProfileReferenceMissing }

public sealed record EmployerApplicationDto(
    Guid Id, string JobId, Guid CandidateId, string? CandidateName, string? CoverLetter,
    ApplicationStatus Status, DateTimeOffset SubmittedAtUtc, DateTimeOffset UpdatedAtUtc,
    string? CurrentCvUrl, CurrentCvStatus CvStatus);
